identity/identity.go

```diff
@@ -147,6 +149,11 @@ func read(repo repository.Repo, ref string) (*Identity, error) {
 
 		// tag the version with the commit hash
 		version.commitHash = hash
+
+		if err := version.Validate(); err != nil {
```

Comment:
hmm, this is done already in `identity_actions.go:MergeAll()` and `Identity.Commit`. It should be enough.